tests/lib/validator_test.py

```diff
@@ -18,104 +18,122 @@
 import pytest
 
 import libnmstate
+from libnmstate import schema
+from libnmstate import state
 from libnmstate.error import NmstateValueError
 
 
 class TestLinkAggregationState(object):
     def test_bonds_with_no_slaves(self):
-        config = [
-            {
-                'name': 'bond0',
-                'link-aggregation': {
-                    'slaves': []
+        desired_state = state.State({
+            schema.Interface.KEY: [
+                {
+                    'name': 'bond0',
+                    'link-aggregation': {
+                        'slaves': []
+                    },
                 },
-            },
-            {
-                'name': 'bond1',
-                'link-aggregation': {
-                    'slaves': []
-                },
-            }
-        ]
-        libnmstate.validator.verify_link_aggregation_state(config, {})
+                {
+                    'name': 'bond1',
+                    'link-aggregation': {
+                        'slaves': []
+                    },
+                }
+            ]
+        })
+
+        libnmstate.validator.verify_link_aggregation_state(desired_state,
+                                                           state.State({}))
```

Comment:
How about an empty_state fixture? The parens and curly braces hurt my eyes. (not a must)

Reply:
Using a fixture here feels an overuse of fixtures.
These are the options, please let me know what you prefer (the first are locals to the test module):
1. state.State(EMPTY_STATE)
2. empty_state()
3. state.empty_state()
4. state.State.empty()

Reply:
`state.State()` could just return an empty state, i.e. use an empty dict by default in the constructor.

Reply:
I am trying to avoid defaults as much as possible as a rule, in this case it is pretty obvious this is a corner case and making it implicit does not make sense.
I took option 2 for now.

BTW, this now shouts that additional unit tests are needed.

Reply:
Why are defaults bad?

Reply:
Also IMHO most Python developers would expect `state.State()` to be an empty state just as `dict()` is an empty dictionary or `list()` an empty list...

Reply:
> Why are defaults bad?

Explicit vs implicit.
Default hide assumptions the coder had. Sometimes they make sense to most, sometimes it makes sense only to some, sometime it makes sense only for one.
As someone who reads the code, I would not expect anyone to create an 'empty' state.